### ISSUE TITLE
Can docxtpl embed one new document in an existing document dynamically? 

### ISSUE THREAD

**Issue by tinhw** [elapouya/python-docx-template#129]
I have read the post : https://github.com/elapouya/python-docx-template/issues/87, it can only use replace method to embed doc, but can not dynamic insert a few documents.
my requirement is that I need insert a few ( uncertain quantities ) documents into an existing one dynamically.

can you help me ? 
Thank you very much.

**Comment by elapouya** [comment]
What is "a few" for you ?

**Comment by qwertypi** [comment]
I'm doing something similar. I have a table on the first page, and each row in the table has a corresponding page. This page is generated from another page template. I'm able to do this following the subdoc example. some code snippets are below.

BUT: I'm getting two newlines at the start of every page after the first one, and don't know how to get rid of them. I've tried putting the whole template in one line, and somewhat randomly adding "p" or "r" without it helping. Any ideas on this?

It may be nice to be able to initialize a Subdoc with an existing docx Document in the init.

Relevant code snippets below:

In template: The first page ends with a page break, then has:
```
{% for page in pages %}
{{p page}}
{% endfor %}
```
The page template also ends with a page break.

In the code:
```
doc = DocxTemplate("template.docx")

pages = []
for idx, data in enumerate(dataset):
	page_doc = DocxTemplate("page_template.docx")

    page_doc.render(context=data)

    sd = doc.new_subdoc()
    sd.subdocx = page_doc.docx

    pages.append(sd)

context = {"pages": pages}
doc.render(context)

doc.save("final.docx")
```

**Comment by elapouya** [comment]
Could you try :

{%p for page in pages %}
{{page}}
{%p endfor %}



Le jeu. 13 sept. 2018 à 21:42, Thomas Kuo <notifications@github.com> a
écrit :

**Comment by qwertypi** [comment]
That combination does the weird thing of removing the page break of the first page, while also NOT getting rid of the starting returns after the first added page.

**Comment by vdsbenoit** [comment]
Any updates on this issue? @qwertypi did you find a solution to get rid of the line breaks?

**Comment by qwertypi** [comment]
@vdsbenoit I don't think I ever explicitly got rid of the line breaks, but I did make them less of an issue. In the templates, I had a Page Break, before the for loop (and a Page Break) at the end of the sub doc. I flipped it such that the page break has at the start of the sub-doc and removed the one in the master template).

This moved the line breaks before the page break where it was less of a nuisance, and also meant I didn't have an empty last page.
